extensions/fluxninja/extconfig/extconfig.go

```diff
@@ -59,3 +65,24 @@ func provideConfig(unmarshaller config.Unmarshaller) (*FluxNinjaExtensionConfig,
 	}
 	return &extensionConfig, nil
 }
+
+func provideEtcdConfigOverride(unmarshaller config.Unmarshaller) (*etcd.EtcdConfigOverride, error) {
```

Comment:
Nit: I think this should take `*FluxNinjaExtensionConfig` as argument, so we avoid parsing config twice